reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/InternalSdkCoreContextTest.kt

```diff
@@ -0,0 +1,625 @@
+import com.datadog.android._InternalProxy
+import com.datadog.android.api.context.DeviceType
+import com.datadog.android.api.context.NetworkInfo
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.stub.StubStorageBackedFeature
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.core.configuration.Configuration
+import com.datadog.android.core.integration.tests.MockServerTest
+import com.datadog.android.core.integration.tests.forge.factories.ConfigurationCoreForgeryFactory
+import com.datadog.android.core.integration.tests.utils.clientToken
+import com.datadog.android.core.integration.tests.utils.env
+import com.datadog.android.core.integration.tests.utils.getFirstPartyHostsWithHeaderTypes
+import com.datadog.android.core.integration.tests.utils.isCrashReportsEnabled
+import com.datadog.android.core.integration.tests.utils.isDeveloperModeEnabled
+import com.datadog.android.core.integration.tests.utils.removeRandomEntries
+import com.datadog.android.core.integration.tests.utils.service
+import com.datadog.android.core.integration.tests.utils.site
+import com.datadog.android.core.integration.tests.utils.variant
+import com.datadog.android.core.thread.FlushableExecutorService
+import com.datadog.android.privacy.TrackingConsent
+import com.datadog.android.trace.TracingHeaderType
+import com.datadog.tools.unit.forge.exhaustiveAttributes
+import com.datadog.tools.unit.forge.useToolsFactories
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.annotation.StringForgeryType
+import fr.xgouchet.elmyr.junit4.ForgeRule
+import fr.xgouchet.elmyr.jvm.useJvmFactories
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.data.Offset
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.File
+import java.util.UUID
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+/**
+ * Provides the tests for the InternalSdkCore API not related with the writing operations.
+ */
+@RunWith(AndroidJUnit4::class)
+class InternalSdkCoreContextTest : MockServerTest() {
+
+    @get:Rule
+    var forge = ForgeRule().useJvmFactories().useToolsFactories().withFactory(ConfigurationCoreForgeryFactory())
+
+    @StringForgery(type = StringForgeryType.ALPHABETICAL)
+    lateinit var fakeUserId: String
+
+    @StringForgery(regex = "[A-Z][a-z]+ [A-Z]\\. [A-Z][a-z]+")
+    lateinit var fakeUserName: String
+
+    @StringForgery(regex = "[a-z]+\\.[a-z]+@[a-z]+\\.[a-z]{3}")
+    lateinit var fakeUserEmail: String
+    private var fakeUserAdditionalProperties: Map<String, Any?> = emptyMap()
+    private lateinit var stubFeature: Feature
+
+    @StringForgery(type = StringForgeryType.ALPHABETICAL)
+    lateinit var fakeFeatureName: String
+
+    private lateinit var fakeTrackingConsent: TrackingConsent
```

Comment:
minor: you can simply add `Forgery` annotation here and remove `lateinit` and `fakeTrackingConsent = forge.aValueFrom(TrackingConsent::class.java)` line.